packages/wallet-backend-update/src/incomingPayment/validation.ts

```diff
@@ -0,0 +1,25 @@
+import { z } from 'zod'
+
+export const incomingPaymentSchema = z.object({
+  body: z.object({
+    paymentPointerId: z.string().uuid(),
+    amount: z.number().positive(),
+    description: z.string().optional()
+  })
+})
+
+export const paymentDetailsSchema = z.object({
+  query: z.object({
+    url: z
+      .string()
+      .url()
+      .regex(
+        new RegExp(
+          /\/[a-z1-9_]*\/incoming-payments\/[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/
```

Comment:
I think we should decode the query parameter first.

Reply:
?

Reply:
Since the payment URL will come as a query parameter, some characters will be encoded. If `express` does not automatically decode them, we should decode them (e.g.: in our use case it will be a payment URL, so the query parameter will be encoded to `http%3A%2F%2Flocalhost%2Fincoming...`).

Reply:
@dragosp1011  can you check this?

Reply:
@raducristianpopa I think it is decoded as it is working now